perfzero/lib/cloud_manager.py

```diff
@@ -160,34 +132,93 @@ def create(username, project, zone, machine_type, accelerator_count,
-
-    cmd = 'gcloud compute ssh {} --project={} --zone={} --command="sudo usermod -a -G docker $USER"'.format(  # pylint: disable=line-too-long
-        instance_name, project, zone)
-    run_command(cmd, is_from_user=True)
-    logging.info('Successfully added user to the docker group\n')
-
-  cmd = 'gcloud compute ssh {} --project={} --zone={} -- -L 6006:127.0.0.1:6006'.format(  # pylint: disable=line-too-long
-      instance_name, project, zone)
+    return
+
+  remote_cmd = 'git clone https://github.com/tensorflow/benchmarks.git'
+  run_command(_ssh_cmd(instance_name, project, zone, remote_cmd),
+              is_from_user=True)
+  logging.info('Successfully checked-out PerfZero code on the '
+               'computing instance\n')
+
+  remote_cmd = 'sudo usermod -a -G docker $USER'
+  run_command(_ssh_cmd(instance_name, project, zone, remote_cmd),
+              is_from_user=True)
+  logging.info('Successfully added user to the docker group\n')
+
+  port_forwarding_cmd = (_ssh_prefix(instance_name, project, zone) +
+                         ' -- -L 6006:127.0.0.1:6006')
   logging.info('Run the command below to ssh to the instance together with '
-               'port forwarding for tensorboard:\n%s\n', cmd)
+               'port forwarding for tensorboard:\n%s\n', port_forwarding_cmd)
+
+
+def _instance_names(instance_name, num_instances):
+  if num_instances == 1:
+    return [instance_name]
+  else:
+    return ['%s-%d' % (instance_name, i) for i in range(num_instances)]
+
+
+def _instances_cmd(command, instance_names, project, zone):
+  """Return a string 'gcloud compute instances' command.
+
+  Args:
+    command: command that goes with 'gcloud compute instances'.
+    instance_names: list of instance names.
+    project: project name
+    zone: GCP zone
+
+  Returns:
+    string command
+  """
+  return 'gcloud compute instances {} {} --project={} --zone={}'.format(
+      command, ' '.join(instance_names), project, zone)
+
+
+def create(instance_name, project, zone, machine_type, accelerator_count,
+           accelerator_type, image, nvme_count, num_instances):
+  """Create gcloud computing instance.
+
+  Args:
+    instance_name: instance_name
+    project: project name
+    zone: zone of the GCP computing instance
+    machine_type: the machine type used for the instance
+    accelerator_count: the number of pieces of the accelerator to attach to
+                       the instance
+    accelerator_type: the specific type of accelerator to attach to the instance
+    image: the name of the image that the disk will be initialized with
+    nvme_count: the number of NVME local SSD devices to attach to the instance
+  """
+  instance_names = _instance_names(instance_name, num_instances)
+  logging.debug('Creating gcloud computing instances {}'.format(instance_names))
+  machine_type = get_machine_type(machine_type, accelerator_count)
+  cmd = (_instances_cmd('create', instance_names, project, zone) +
```

Comment:
Let's say user runs `cloud_manager.py create --num_instances=2` to create two instances. Then user runs `cloud_manager.py create --num_instances=3`. Does the second command fail, or does it create one extra instance? Either way, it is probably hard to achieve the ideal goal of creating one extra instance for user.